### ISSUE TITLE
Unable to login with SSO when linking to app.builbuddy.io from a sandboxed iframe missing allow-forms

### ISSUE THREAD

**Issue by norrs** [buildbuddy-io/buildbuddy#7684]
**Describe the bug**

> Blocked form submission to '' because the form's frame is sandboxed and the 'allow-forms' permission is not set.

Browser console in developer console complains about this. 

**To Reproduce**

`index.html` : 

```html
<html>
 <iframe class="lens-container"
   style="visibility: visible; height: 1192px;"
   id="iframe-1"
   sandbox="allow-scripts allow-top-navigation allow-popups allow-same-origin" data-lens-index="1" data-lens-name="buildlog" src="foo.html">
</iframe>
</html>
````

`foo.html`:

```html
<html>
  <body>
    <a target="_blank" href="https://app.buildbuddy.io">https://app.buildbuddy.io</a>
  </body>
</html>
```

* Open `index.html` on your PC
* click the link which spawns a new tab with `app.builbuddy.io`
* you are unable to click the `Continue with SSO`. 

**Expected behavior**

Be able to insert out team slug which enables vendor specific SSO. 

**Screenshots**

![image](https://github.com/user-attachments/assets/96d358db-3e46-4625-a66c-5fc24e801a33)

**Desktop (please complete the following information):**

- OS: macos
- Browser: Chrome
- Version: Version 129.0.6668.90 (Official Build) (arm64)

**Additional context**

**Comment by norrs** [comment]
My provided example works now 🤯 , which is odd as I did try this in incognito as well I believe. 🤔 

but I also found another way to reproduce the same exact error,  that I cannot easily share. Im asking internally within my organization if I can share my method to reproduce this error over a video call.  

Would you be available over a video call if I get an ok to do this @sluongng ?

**Comment by sluongng** [comment]
Yes can do.

You can also send a recording to my work email son@buildbuddy.io or join our community Slack here https://community.buildbuddy.io/ and send me a private message.

**Comment by norrs** [comment]
Ok, im still awaiting for the OK on my side to be able to join a videocall. 

I did some further digging on how I'am able to reproduce this; and it probably is pebkab stuff at our side but I find it interesting:

* We use k8s internal tooling for CI, prow : https://github.com/kubernetes-sigs/prow
* It uses something called spyglass to embed things to prow ... https://docs.prow.k8s.io/docs/spyglass/

* Prow uses iframes to embed "build logs" from our bazel targets ..  (which is executed on buildbuddy runners)

Here is the funny part, even if this build log has a `<a href=buildbuddy log target=_blank> and you click this link and it opens a new `tab`, it seems to still be tracked to be part of the sandboxing of the iframe from the other tab (prow) where I tried to open buildbuddy from. 

Im not sure if this is intended browser logic or not. 

the iframe has `sandbox="allow-scripts allow-top-navigation allow-popups allow-same-origin"` decorated. 

So when I dont have a session, it shows the new tab with app.buildbuddy.io and login prompt (makes sense), but trying to click Continue with SSO spams the `Blocked form submission to '' because the form's frame is sandboxed and the 'allow-forms' permission is not set.`. 

Best part of "understanding" this, if I keep using this newly open tab, and continue browsing the internet with it, and comes back to try to login to app.buildbuddy.io in the same tab, and even hard refresh with invalidate cache, it still remember the state of this special tab and it complains about  `blocked form submission due to being sandboxed` ...

**Comment by norrs** [comment]
Managed to create a local reproducible POC:

`index.html` : 

```html
<html>
 <iframe class="lens-container"
   style="visibility: visible; height: 1192px;"
   id="iframe-1"
   sandbox="allow-scripts allow-top-navigation allow-popups allow-same-origin" data-lens-index="1" data-lens-name="buildlog" src="foo.html">
</iframe>
</html>
````

`foo.html`:

```html
<html>
  <body>
    <a target="_blank" href="https://app.buildbuddy.io">https://app.buildbuddy.io</a>
  </body>
</html>
```

Open `index.html` on your PC, then try to click the link which spawns a new tab with `app.builbuddy.io`, and you are unable to click the `Continue with SSO`. 

Questions:

 1) is there security concerns with adding `allow-forms` in the iframe?
 
 2) How can a `<a href=.. target=_blank>aaaa</a>` which spawns a new tab not be a new sandbox? 
 
  How is Chrome's sandboxing working? Is there any special rules in regards of iframes? 
  .. aren't a new tab a new process, which again is a new sandbox? 
  .. is it special sandboxing rules in play when links are opened from iframe? 
  
  Do we have any browser experts lurking around? 😅

**Comment by fmeum** [comment]
I'm certainly not a browser expert, but I hope that I can still provide some clarifications :-)

> .. aren't a new tab a new process, which again is a new sandbox?

Whether or not tabs use different processes is an implementation detail of the browser that aims to contain the impact of vulnerabilities in the browser. The `<iframe>` sandbox is a high-level concept that's part of the HTML spec. The two thus aren't related.

> How can a <a href=.. target=_blank>aaaa</a> which spawns a new tab not be a new sandbox?

By default, the `<iframe>` sandbox is sticky and "follows" any navigation that originates from the original sandboxed document. You can control this via the (confusingly named) `[allow-popups-to-escape-sandbox`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#allow-popups-to-escape-sandbox) directive for the `sandbox` attribute.

> is there security concerns with adding allow-forms in the iframe?

The answer is "it depends". Generally speaking, an untrusted `iframe` with fewer sandbox restrictions clearly has more ways to perform malicious actions. In this particular case though, you already trust it with script execution and it's not an arbitrary site, it's the frontend to the service you entrust your builds (metadata) to. :-)

**Comment by norrs** [comment]
I agree; I made a pull request upstream (https://github.com/kubernetes-sigs/prow/pull/294) for the CI and tool we use. If we configure our tool to generate links to buildbuddy from within an iframe, we should ensure we populate the required properties for the sandbox to run.  :+1:  

https://web.dev/articles/sandboxed-iframes had a good demo on understanding these concepts about sandboxing iframe, and it's joyful to finally found the cause of the issue on our side. 

I took the liberty to update the title and description of the "bug", in case someone stumbles upon this via a google search or something similar. 

I suppose we could close this issue here @fmeum . Thanks for the rubberdducking on slack @sluongng . :-)
